### PR TITLE
Make cuda conflicts useful for cray too

### PR DESCRIPTION
**Pull request by haampie**
The CUDA compiler conflicts are valid on Cray too, and likely on Darwin x86_64 with %gcc and %clang in case you'd build these compilers instead of using apple-clang, so drop platform=linux in the conflicts entirely.

Also add a note about missing conflicts for %apple-clang on Darwin; the comment there was a bit outdated.

This PR is not meant to add more conflicts for %apple-clang, since I don't have access to these macos+nvidia machines at all.

### CODE REVIEW

**Comment by naromero77** [comment]
A number of comments where removed which is OK if they are no longer relevant. It would be good to refresh the comments so someone else could understand what was done and why.

**Comment by haampie** [comment]
I thought the 'difficult to express with the current Spack conflict syntax' comment applied mostly to `clang` being different on Darwin, which was solved since we now have `clang` and `apple-clang` separately.

And I removed

> There are at least three cases to be aware of for compiler conflicts
> 1. Linux x86_64
> 2. Linux ppc64le
> 3. Mac OS X

because I dropped `platform=linux` from the conflicts, and the 3 sections are rather obvious in the file, and the linux and powerpc sections link to the sources in a comment already. Let me know if you want to keep these comments.

**Review by ax3l — DISMISSED** [review]
LGTM, I guess if someone was interested to add old CUDA macOS conflicts the would add them for `apple-clang` only anyway.

**Comment by haampie** [comment]
Awaiting @naromero77's approval

**Comment by naromero77** [comment]
I don't think we need to restore the old comments. I still think its desirable to add a comment saying why the conflict use `target` instead of `platform` as it wouldn't be obvious from to a new person.

**Comment by haampie** [comment]
@naromero77 do you think the comment from the last commit explains it well enough?

**Review by naromero77 — APPROVED** [review]
Looks good to me.

I think @ax3l has already approved.

I am going on vacation, someone with merge privilege will need to merge it in.

**Review by alalazo — APPROVED** [review]

**Pull request merged**
